server/app.mjs

```diff
@@ -324,4 +343,79 @@ const dismount = (socket) => {
     socket.removeAllListeners("confirmResponse");
     socket.removeAllListeners("userMessage");
     socket.removeAllListeners("disconnect");
-}
\ No newline at end of file
+}
+
+const initGPTScriptConfig = async (gptscript) => {
+    // Run a non-LLM no-op script to ensure the GPTScript config exists
+    const run = await gptscript.evaluate({
+        instructions: '#!sys.echo noop'
+    })
+    await run.text()
+
+    const configPath = gptscriptConfigPath();
+
+    fs.readFile(configPath, 'utf8', (err, data) => {
+        if (err) {
+            console.error('Error reading config file:', err);
+            return;
+        }
+
+        let config;
+        try {
+            config = JSON.parse(data);
+        } catch (parseErr) {
+            throw new Error(`Error parsing config file: ${parseErr}`);
+        }
+
+        // Default values to add if they don't exist
+        const defaultConfig = {
+            gatewayURL: 'https://gateway-api.gptscript.ai',
+            integrations: {
+                microsoft365: 'microsoft365',
+                slack: 'slack',
+                notion: 'notion'
+            }
+        };
+
+        // Update the config object with default values if they don't exist
+        config = {
+            ...defaultConfig,
+            ...config,
+            integrations: {
+                ...defaultConfig.integrations,
+                ...config.integrations
+            }
+        };
+
+        // Write the updated config back to the file
+        fs.writeFile(configPath, JSON.stringify(config, null, 2), 'utf8', (writeErr) => {
+            if (writeErr) {
+                console.error('Error writing config file:', writeErr);
+                return;
+            }
+            console.log('Config file updated successfully');
+        });
+    });
+}
+
+function gptscriptConfigPath() {
+    const homeDir = os.homedir();
+    let configPath;
+
+    if (os.platform() === 'darwin') {
+        // macOS
+        configPath = path.join(homeDir, 'Library', 'Application Support', 'gptscript', 'config.json');
```

Comment:
This can still be overridden by `XDG_CONFIG_HOME`, even on macOS. (I've done it, for example). The `xdg` library we use in Go honors this, so I think we should check for it here too.

Reply:
I updated the logic to account for `XDG_CONFIG_HOME` on macos.